src/geom/elem.C

```diff
@@ -3488,7 +3488,13 @@ BoundingBox Elem::loose_bounding_box () const
   return BoundingBox(pmin, pmax);
 }
 
-
+Point
+Elem::side_vertex_average_normal(const unsigned int /*s*/) const
+{
+  libmesh_not_implemented_msg("Side normals not implemented for element of type: " <<
```

Comment:
We really ought to be able to implement a default here.  Give a side `FE` a `QTrap`, call `get_normals()`, average, renormalize.  Aside from implementing all the higher-order elements at once, that would let us add a unit test comparing the base class vs subclass implementations, and would catch things like forgetting to renormalize before averaging in a subclass.

Reply:
`QTrap` because it looked like your code was trying to do the side-vertex average-normal calculation, and `QTrap` would let you get that.  If we want to do the side-vertex-average normal instead (which would be reasonable, if we can get it right quickly on Quad4) then `QGauss` (but hard-coded to `CONSTANT` or `FIRST`, and either way we should have p refinement disabled in the `FE`) would get that.

Reply:
I ll try QTrap. 
I think the average might be preserved on quad4, which is the only 'nonlinear' case I have tested

Reply:
QTrap gets more than one unit normal. QTrap is a fixed size right? I need only one face normal. Should I average ?

Reply:
so way forward is likely:
high order quadrature for Elem.C implementation and average the results

I ll try that

Reply:
yeah I think this (once corrected) is the better way

```
  const std::unique_ptr<const Elem> face = this->build_side_ptr(s);
  std::unique_ptr<libMesh::FEBase> fe(
      libMesh::FEBase::build(dim, libMesh::FEType(this->default_order())));
  const std::vector<Point> & normals = fe->get_normals();
  fe->reinit(this, s, {face->vertex_average()});
  return _normals[0];
```

Reply:
I think we are linear on the edges of a quad4. so what I have is correct for quad4 side vertex normals.

Now in the middle of a quad4 face of a hex8, I need to check

Reply:
so at eta=0, xi = 0, we are at the center of the ref element, and also
XY = 1/4 x11y11 +  1/4 x21y21 +  1/4 x12y12  +  1/4 x22y22 , the side vertex average in physical space

Reply:
so I think this solves our "reference" elem.C implementation? At least for linear lagrange mappings

for quad9 it's a little different. 
We map to the middle node from setting xi = 0, eta = 0. which may not be the vertex average

Reply:
For the general case I switch to using the master element vertex average. That should be well defined for all element types, except polys but there's a derived iimplementation for them

Reply:
Yeah, you're quite right.  Quad4 always works, even twisted in 3D.  There might be other tricks that work in higher order but probably nothing worth overriding your new `Elem::` default for.